### DISCUSSION TITLE
Events/Function trigger for realtime subscribe/unsubscribe

### DISCUSSION THREAD

**Discussion by KimCalvin** [appwrite/appwrite#2587] [Ideas]
In my opinion it would be useful to be able to execute a function based onOpen and onClose in [realtime.php](https://github.com/appwrite/appwrite/blob/1a0f740459fde58dc41112861f47bc06d1ae5c4c/app/realtime.php).

I'd suggest adding the [Events](https://github.com/appwrite/appwrite/blob/1a0f740459fde58dc41112861f47bc06d1ae5c4c/app/config/events.php) "realtime.subscription.open" and "realtime.subscription.close" and making them triggers for cloud functions.

This would for example allow for easy [user presence](https://firebase.google.com/docs/firestore/solutions/presence) management depending on the users open socket connection.

**Comment by stnguyen90** [comment]
This sounds like this feature request: https://github.com/appwrite/appwrite/issues/2418. If so, can this be closed in favor of that issue?
